### PR TITLE
Use quarkus.properties to get versions in codestarts and make them required

### PR DESCRIPTION
**Pull request by ia3andy**
Supersed https://github.com/quarkusio/quarkus/pull/14481

cc @gsmet

### CODE REVIEW

**Comment by ia3andy** [comment]
I will add a test

File: devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/maven/codestart.yml [hunk 1]
```diff
@@ -14,7 +14,6 @@ language:
         version: 3.8.1
       maven-surefire-plugin:
         version: 2.22.1
```
Comment (gsmet) [inline devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/maven/codestart.yml:16]: But why do we still have the old versions there?

Comment (ia3andy) [reply devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/maven/codestart.yml:16]: @gsmet they are just default, I didn't start from your PR. But with this fix, the default will be overrided by the data from the `quarkus.properties`.

**Review by ia3andy — COMMENTED** [review]
@famod in the end, it's what it does.. it just that I prefer having one property file filtered than the codestart itself which could lead to unpredicted behaviors.. 

This fix is just plugging the same data as the legacy was using..

I will soon remove the legacy mapper and make this a bit cleaner/simpler when I find time for this.. it's not a priority since it works..

**Comment by famod** [comment]
@ia3andy Ok. I was just thinking that with plain Maven resource filtering you wouldn't need to define those properties in a java class.

File: devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/maven/codestart.yml (hunk 1 above)
Comment (gsmet) [reply devtools/platform-descriptor-json/src/main/resources/codestarts/quarkus/core/buildtool/maven/codestart.yml:16]: OK, then, let's merge both PRs.

**Comment by gsmet** [comment]
Yeah, the difference between what you do and what @famod would want is that with resource filtering, we would break things if we miss something. And I wonder if it wouldn't be better.

With what you do, if we break things, it will just go back silently to default values that will probably be outdated.

**Comment by ia3andy** [comment]
> Yeah, the difference between what you do and what @famod would want is that with resource filtering, we would break things if we miss something. And I wonder if it wouldn't be better.
> 
> With what you do, if we break things, it will just go back silently to default values that will probably be outdated.

Then we should just remove those defaults (to make them required), it would fail if they aren't set from above. 

TBH I would be scared of mixing qute templates + maven syntax + resource filtering  😅

**Comment by ia3andy** [comment]
@gsmet you can close your PR, I removed the default, they now are required..

**Comment by ia3andy** [comment]
ping @gsmet :)

**Review by gsmet — APPROVED** [review]

**Pull request merged**